SU2_CFD/src/solvers/CFlameletSolver.cpp

```diff
@@ -0,0 +1,890 @@
+      SlidingStateNodes[iMarker].resize(nVertex[iMarker],0);
+    }
+  }
+
+  /*-- Allocation of inlets has to happen in derived classes (not CTurbSolver),
+    due to arbitrary number of turbulence variables ---*/
+
+  Inlet_ScalarVars.resize(nMarker);
+  for (unsigned long iMarker = 0; iMarker < nMarker; iMarker++) {
+    Inlet_ScalarVars[iMarker].resize(nVertex[iMarker],nVar);
+    for (unsigned long iVertex = 0; iVertex < nVertex[iMarker]; ++iVertex) {
+      for (unsigned short iVar = 0; iVar < nVar; iVar++){
+        Inlet_ScalarVars[iMarker](iVertex,iVar) = Scalar_Inf[iVar];
+      }
+    }
+  }
+
+  Inlet_Position = nDim*2+2;
+  if (turbulent) {
+    if (turb_SA) Inlet_Position += 1;
+    else if (turb_SST) Inlet_Position += 2;
+  }
+
+  /*--- Add the solver name (max 8 characters) ---*/
+  SolverName = "FLAMELET";
+
+}
+
+CFlameletSolver::~CFlameletSolver(void) {
+  
+  unsigned long iMarker, iVertex;
+  unsigned short iVar;
+  
+}
+
+void CFlameletSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container, CConfig *config,
+         unsigned short iMesh, unsigned short iRKStep, unsigned short RunTime_EqSystem, bool Output) {
+
+  unsigned long InnerIter = config->GetInnerIter();
+  bool disc_adjoint       = config->GetDiscrete_Adjoint();
+  bool limiter_flow       = ((config->GetKind_SlopeLimit_Flow() != NO_LIMITER) &&
+                            (InnerIter <= config->GetLimiterIter()) && !(disc_adjoint && config->GetFrozen_Limiter_Disc()));
+  bool limiter_scalar     = ((config->GetKind_SlopeLimit_Scalar() != NO_LIMITER) &&
+                            (InnerIter <= config->GetLimiterIter()) && !(disc_adjoint && config->GetFrozen_Limiter_Disc()));
+  
+  /*--- Set the primitive variables ---*/
+  
+  unsigned long ErrorCounter = SetPrimitive_Variables(solver_container, config, Output);
+  
+  /*--- Initialize the Jacobian matrices ---*/
+  
+  if (!disc_adjoint) Jacobian.SetValZero();
+  
+  if (config->GetKind_Gradient_Method() == GREEN_GAUSS) SetSolution_Gradient_GG(geometry, config);
+  if (config->GetKind_Gradient_Method() == WEIGHTED_LEAST_SQUARES) SetSolution_Gradient_LS(geometry, config);
+  
+  /*--- Upwind second order reconstruction ---*/
+  
+  if (limiter_scalar) SetSolution_Limiter(geometry, config);
+  
+  if (limiter_flow) solver_container[FLOW_SOL]->SetPrimitive_Limiter(geometry, config);
+  
+}
+
+void CFlameletSolver::Postprocessing(CGeometry *geometry,
+                                          CSolver **solver_container,
+                                          CConfig *config,
+                                          unsigned short iMesh) { }
+
+unsigned long CFlameletSolver::SetPrimitive_Variables(CSolver **solver_container,
+                                                           CConfig *config,
+                                                           bool Output) {
+  
+  unsigned long iPoint, ErrorCounter = 0;
+  su2double Density, Temperature, lam_visc = 0.0, eddy_visc = 0.0, Cp;
```

Comment:
Continuing the conversation from yesterday.
@bigfooted @danielmayer so far as handling diffusivity for the different solvers, this looks like the solution.
Do it as part of the preprocessing, or in case it is constant it can be set in the constructor.

Reply:
yes, diffusivity is a bit messy now. For combustion, we get diffusivity from a lookup table (just like viscosity, temperature, etc, all read from a file).  For the passive scalar, I want to extend this to a mixing model.

Reply:
Ok each solver can be as messy as it wants in that primitive variables method.
But in the base solver we will avoid logic relating with the kind of solver (if passive if heat if turbulence).

Are you planning to do any more work on CScalarSolver? If not I will use what you have now to create the CFVMSolverBase and apply it to turbulence. I will then create a PR to develop, and after that is merged you can apply it here to the PassiveScalar and FlameletSolvers, @oleburghardt can then do it for the heat solver in #1226.
I think it is the best compromise to keep PR sizes manageable.
Does that sound like a plan?

Reply:
@pcarruscag Please go ahead and do what needs to be done. We will also merge cpassivescalarsolver.cpp and cflameletsolver.cpp as they end up being almost the same. Regarding diffusivity I think now that this should be treated in the same way as viscosity, so as part of the regular fluid model properties. After all, fluids also experience self-diffusion so even for a single component gas it makes sense to have diffusivity as a fluid property.